lib/db/lowlevel.go

```diff
@@ -48,27 +41,55 @@ func init() {
 // any given backend.
 type Lowlevel struct {
 	backend.Backend
-	folderIdx  *smallIndex
-	deviceIdx  *smallIndex
-	keyer      keyer
-	gcMut      sync.RWMutex
-	gcKeyCount int
-	gcStop     chan struct{}
+	folderIdx          *smallIndex
+	deviceIdx          *smallIndex
+	keyer              keyer
+	gcMut              sync.RWMutex
+	gcKeyCount         int
+	gcStop             chan struct{}
+	indirectGCInterval time.Duration
+	recheckInterval    time.Duration
 }
 
-func NewLowlevel(backend backend.Backend) *Lowlevel {
+func NewLowlevel(backend backend.Backend, opts ...Option) *Lowlevel {
 	db := &Lowlevel{
-		Backend:   backend,
-		folderIdx: newSmallIndex(backend, []byte{KeyTypeFolderIdx}),
-		deviceIdx: newSmallIndex(backend, []byte{KeyTypeDeviceIdx}),
-		gcMut:     sync.NewRWMutex(),
-		gcStop:    make(chan struct{}),
+		Backend:            backend,
+		folderIdx:          newSmallIndex(backend, []byte{KeyTypeFolderIdx}),
+		deviceIdx:          newSmallIndex(backend, []byte{KeyTypeDeviceIdx}),
+		gcMut:              sync.NewRWMutex(),
+		gcStop:             make(chan struct{}),
+		indirectGCInterval: indirectGCDefaultInterval,
+		recheckInterval:    recheckDefaultInterval,
+	}
+	for _, opt := range opts {
+		opt(db)
 	}
 	db.keyer = newDefaultKeyer(db.folderIdx, db.deviceIdx)
 	go db.gcRunner()
 	return db
 }
 
+type Option func(*Lowlevel)
+
+// WithRecheckInterval sets the time interval in between metadata recalculations
+// and consistency checks.
+func WithRecheckInterval(dur time.Duration) Option {
+	return func(db *Lowlevel) {
+		if dur != 0 {
```

Comment:
So about that, I would do `STRECHECKDBEVERY=0 syncthing` to force a check *now*. This breaks that, I'd have to say `STRECHECKDBEVERY=1s` instead. Which is maybe fine... At least for the GC interval zero isn't a great choice. (We could say that zero means no GC, but it's also intuitive enough to say `STGCINDIRECTEVERY=100000d` or so...)

Reply:
There's also negative numbers which we might want to handle or reject

Reply:
I handled negative as meaning default too. The reasoning for 0 meaning default is that I don't want to check the options in `lib/syncthing` to decide whether to use the `With` options - just send the zero value and get the default. And I think "properly" adding disabling is overkill, as I don't see a use-case outside of debugging (and there the workaround with huge intervals is fine).